common/src/crs.rs

```diff
@@ -9,12 +11,17 @@ const G1_START: usize = 28;
 const G2_START: usize = 28 + (5_040_000 * 64);
 const G2_END: usize = G2_START + 128 - 1;
 
-fn transcript_location() -> std::path::PathBuf {
-    let mut transcript_dir = dirs::home_dir().unwrap();
-    transcript_dir.push(std::path::Path::new("noir_cache"));
-    transcript_dir.push(std::path::Path::new("ignition"));
-    transcript_dir.push(std::path::Path::new("transcript00.dat"));
-    transcript_dir
+fn transcript_location() -> PathBuf {
+    match env::var("BARRETENBERG_TRANSCRIPT") {
+        Ok(dir) => PathBuf::from(dir),
+        Err(_) => {
+            let mut transcript_dir = dirs::home_dir().unwrap();
+            transcript_dir.push(std::path::Path::new("noir_cache"));
+            transcript_dir.push(std::path::Path::new("ignition"));
+            transcript_dir.push(std::path::Path::new("transcript00.dat"));
+            transcript_dir
+        }
```

Comment:
```suggestion
        Err(_) => dirs::home_dir()
            .unwrap()
            .join("noir_cache")
            .join("ignition")
            .join("transcript00"),
```

I've noticed that we tend to use this mutable "create many paths and push" pattern over "chain joins of strings" in quite a few places. I find this to be more readable so is there a reason why not to use it?

Reply:
Seems fine. I didn't know the API existed but it does the same thing internally, so let's do it.